pip/configuration.py

```diff
@@ -24,12 +24,22 @@
 )
 from pip.utils import ensure_dir, enum
 
+if False:
+    from typing import Any, Dict, Iterable, List, NewType, Optional, Tuple
+
+    try:
+        from configparser import ConfigParser
+    except ImportError:
+        from ConfigParser import ConfigParser  # type: ignore
+
```

Comment:
Personally, "just mypy things" turns this into some sort of unexplained magic that people maintaining the file will cargo cult without understanding. If we're going to use typing, I think it's important that all the maintainers are able to understand it.

See my other comment - maybe we can put all of the "stuff that needs to go at the top of the file" like this into a separate ```piptyping``` module, which has an extensive comment at the top explaining why we are doing this, why pip is a special case, and providing links for anyone who needs to maintain that file, and then just import that one file at the top of each source file that uses annotations.

That way the typing black magic is encapsulated, and doesn't clutter every source file with stuff people aren't expected to understand, while still giving people who *need* to understand it a chance to follow what's going on.